### PR TITLE
Fix ignore inline comment.

### LINKED ISSUE #585: starlette should ignore "inline" comment in env file
Hi,
when using https://www.starlette.io/config/ , if we put 
```
VAR = 'value'   # this var is use for ....
```
we get error related to the expected role of the var.

For example in my case 
```
  File "app.py", line 292, in <module>
    Mount('/files', StaticFiles(directory=settings('JOPLIN_RESOURCES'))),
  File "[...]/lib/python3.7/site-packages/starlette/staticfiles.py", line 55, in __init__
    raise RuntimeError(f"Directory '{directory}' does not exist")
RuntimeError: Directory '/home/foxmask/.config/joplin-desktop/resources/'   # path to get the joplin `resources` to be able to display them' does not exist
```
or if that was for a HTTP port :
```
Traceback (most recent call last):
  File "app.py", line 303, in <module>
    uvicorn.run(main_app, host='0.0.0.0', port=settings('HTTP_PORT', cast=int, default=8001))
  File "[...]lib/python3.7/site-packages/starlette/config.py", line 59, in __call__
    return self.get(key, cast, default)
  File "[...]lib/python3.7/site-packages/starlette/config.py", line 69, in get
    return self._perform_cast(key, value, cast)
  File "[...]lib/python3.7/site-packages/starlette/config.py", line 103, in _perform_cast
    f"Config '{key}' has value '{value}'. Not a valid {cast.__name__}."
ValueError: Config 'HTTP_PORT' has value '8001   # listen port number'. Not a valid int.
```

### PR DESCRIPTION
**Pull request by alaminopu**
Closes #585

### CODE REVIEW

**Comment by blueyed** [comment]
For (clickable, linked) reference: #585

File: starlette/config.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def _read_file(self, file_name: str) -> typing.Dict[str, str]:
                 line = line.strip()
                 if "=" in line and not line.startswith("#"):
                     key, value = line.split("=", 1)
+                    if "#" in value:
+                        value = value.split("#", 1)[0]
```
Comment (blueyed) [inline starlette/config.py:82]: This will fail for `SOME_URL=http://example.com#fragment`.

**Comment by lovelydinosaur** [comment]
Updated the description and title to correctly reference the corresponding issue.

File: starlette/config.py (hunk 1 above)
Comment (lovelydinosaur) [reply starlette/config.py:82]: Indeed, we'd need a smarter approach here.

Comment (alaminopu) [reply starlette/config.py:82]: @blueyed Thanks for pointing out the issue might occur. I will think about a smarter approach here.

**Comment by lovelydinosaur** [comment]
Let's treat this as out of scope, at least for now. See https://github.com/encode/starlette/issues/585#issuecomment-521507083

Happy to accept documentation pull requests, tho.

**Pull request closed without merging**